server/src/controllers/room.js

```diff
@@ -0,0 +1,33 @@
+const Boom = require('@hapi/boom');
+
+const { verifyCookie } = require('../utils/jwt');
+const { roomSchema } = require('./validation/roomSchema');
+const { addNewRoom, getRoom } = require('../database/queries');
+
+const addRoom = (req, res, next) => {
+  const { name } = req.body;
+  roomSchema
+    .validateAsync({ name })
+    .catch((error) => {
+      throw Boom.badRequest(error.message);
+    })
+    .then(() => verifyCookie(req.cookies.token))
```

Comment:
This should be as a middleware, that puts the cookies content in req.user
and you can access user_id and role like req.user.role

Reply:
this makes this pull request pending to omar's #45